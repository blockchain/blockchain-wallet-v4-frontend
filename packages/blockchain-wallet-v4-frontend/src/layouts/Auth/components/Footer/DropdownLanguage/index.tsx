import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import Cookies from 'universal-cookie'

import { SimpleDropdown } from 'blockchain-info-components/src/Dropdowns'
import { preferences } from 'data/actions'
import { getLanguage } from 'data/preferences/selectors'
import { languagesSortedByName as languages } from 'services/locales'

const DropdownLanguage = ({ color, size }: { color: string; size: string }) => {
  const dispatch = useDispatch()
  const currentLanguage = useSelector(getLanguage)

  const handleClick = (selectedLanguage) => {
    const cookies = new Cookies()
    dispatch(preferences.setLanguage({ language: selectedLanguage.language, showAlert: true }))
    cookies.set('clang', selectedLanguage.language, {
      domain: '.blockchain.com',
      path: '/'
    })
  }

  const languageList = languages.map((lang) => {
    return {
      language: lang.language,
      text: lang.name,
      value: lang.language
    }
  })

  return (
    <SimpleDropdown
      color={color || 'whiteFade900'}
      uppercase={false}
      items={languageList}
      selectedValue={currentLanguage}
      size={size}
      callback={(selectedLanguage) => handleClick(selectedLanguage)}
    />
  )
}

export default DropdownLanguage
