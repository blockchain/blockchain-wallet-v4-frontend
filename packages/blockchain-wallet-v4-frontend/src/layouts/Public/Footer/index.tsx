import React from 'react'

import { FormattedMessage } from 'react-intl'
import { Link } from 'blockchain-info-components'
import { LinkContainer } from 'react-router-bootstrap'
import styled from 'styled-components'

const VersionWrapper = styled.span`
  margin-top: 4px;
  margin-left: 16px;
  margin-right: 16px;
`

// @ts-ignore
const V = APP_VERSION

const Footer = () => (
  <>
    <VersionWrapper>
      <Link
        color='grey400'
        href='https://github.com/blockchain/blockchain-wallet-v4-frontend/releases'
        size='16px'
        target='_blank'
        weight={500}
      >
        Version {V}
      </Link>
    </VersionWrapper>
    <LinkContainer to='/help'>
      <Link size='16px' weight={600} color='grey400' data-e2e='loginGetHelp'>
        <FormattedMessage
          id='scenes.login.needhelp'
          defaultMessage='Need some help?'
        />
      </Link>
    </LinkContainer>
  </>
)

export default Footer
