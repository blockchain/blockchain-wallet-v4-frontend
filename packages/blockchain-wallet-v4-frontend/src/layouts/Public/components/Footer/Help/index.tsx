import React from 'react'
import { FormattedMessage } from 'react-intl'
import { LinkContainer } from 'react-router-bootstrap'
import styled from 'styled-components'

import { Text } from 'blockchain-info-components'

const HelpContainer = styled.div`
  display: inline-flex;
  cursor: pointer;
`

const Help = (props) => {
  return (
    <HelpContainer>
      <LinkContainer to={props.authProduct === 'WALLET' ? '/help' : '/help-exchange'}>
        <Text color='grey400' size='16px' weight={500}>
          <FormattedMessage id='scenes.login.need.help' defaultMessage='Need Help?' />
        </Text>
      </LinkContainer>
    </HelpContainer>
  )
}
export default Help
