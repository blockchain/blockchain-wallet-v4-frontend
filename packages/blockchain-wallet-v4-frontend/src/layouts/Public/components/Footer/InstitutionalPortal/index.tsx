import React from 'react'
import { FormattedMessage } from 'react-intl'
import { connect } from 'react-redux'
import { LinkContainer } from 'react-router-bootstrap'
import { bindActionCreators } from 'redux'
import styled from 'styled-components'

import { Text } from 'blockchain-info-components'
import { actions } from 'data'
import { LOGIN_FORM } from 'data/auth/model'
import { LoginSteps } from 'data/types'

const SubCard = styled.div`
  display: flex;
  justify-content: center;
  cursor: pointer;
`
const SignUpText = styled(Text)`
  cursor: pointer;
  &:hover {
    color: ${(props) => props.theme.white};
    font-weight: 600;
  }
`

const InstitutionalPortal = (props) => (
  <LinkContainer to='/login?product=exchange&userType=institutional'>
    <SubCard
      onClick={() => props.formActions.change(LOGIN_FORM, 'step', LoginSteps.INSTITUTIONAL_PORTAL)}
    >
      <Text size='16px' color='grey400' weight={500}>
        <FormattedMessage
          id='layouts.wallet.footer.institutional_portal'
          defaultMessage='Are you an institutional user?'
        />
      </Text>
      &nbsp;
      <SignUpText size='16px' color='white' weight={600}>
        <FormattedMessage id='scenes.login.wallet.exchange_login' defaultMessage='Log In ->' />
      </SignUpText>
    </SubCard>
  </LinkContainer>
)

const mapDispatchToProps = (dispatch) => ({
  formActions: bindActionCreators(actions.form, dispatch)
})

export default connect(null, mapDispatchToProps)(InstitutionalPortal)
