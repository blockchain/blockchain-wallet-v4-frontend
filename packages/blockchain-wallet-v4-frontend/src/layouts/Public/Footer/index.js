import React from 'react'
import styled from 'styled-components'
import { FormattedMessage } from 'react-intl'

import { Image, Link } from 'blockchain-info-components'
import DropdownLanguage from 'components/DropdownLanguage'

const Wrapper = styled.footer`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  min-height: 60px;
`
const LinkContainer = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;

  & > a {
    padding: 10px;
  }
`

const RightContainer = styled.div`
  margin: 20px 50px 0 0;
  display: flex;
  flex-direction: column;
  align-items: flex-end;

  & > :last-child {
    padding: 5px;
  }
`

const Footer = () => {
  return (
    <Wrapper>
      <LinkContainer>
        <Image name='blue-logo' height='40px' width='40px' />
        <Link
          href='https://blockchain.com/explorer'
          size='12px'
          weight={300}
          color='white'
          uppercase
        >
          <FormattedMessage
            id='layouts.public.footer.explorer'
            defaultMessage='Data'
          />
        </Link>
        <Link
          href='https://blockchain.com/about'
          size='12px'
          weight={300}
          color='white'
          uppercase
        >
          <FormattedMessage
            id='layouts.public.footer.about'
            defaultMessage='About'
          />
        </Link>
        <Link
          href='https://blog.blockchain.com'
          size='12px'
          weight={300}
          color='white'
          uppercase
        >
          <FormattedMessage
            id='layouts.public.footer.blog'
            defaultMessage='Blog'
          />
        </Link>
        <Link
          href='https://support.blockchain.com'
          size='12px'
          weight={300}
          color='white'
          uppercase
        >
          <FormattedMessage
            id='layouts.public.footer.support'
            defaultMessage='Support'
          />
        </Link>
      </LinkContainer>
      <RightContainer>
        <DropdownLanguage color='white' />
        <Link
          href='https://github.com/blockchain/blockchain-wallet-v4-frontend/releases'
          target='_blank'
          size='12px'
          weight={200}
          color='white'
        >
          Version {APP_VERSION}
        </Link>
      </RightContainer>
    </Wrapper>
  )
}

export default Footer
