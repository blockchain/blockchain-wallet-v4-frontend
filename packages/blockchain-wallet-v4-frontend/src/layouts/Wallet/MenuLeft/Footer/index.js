import React from 'react'
import { connect } from 'react-redux'
import { compose } from 'redux'
import { withRouter } from 'react-router-dom'

import { getData } from './selectors'
import Footer from './template.js'

class FooterContainer extends React.PureComponent {
  render () {
    const { ...props } = this.props
    return <Footer {...props} />
  }
}

const mapStateToProps = state => ({
  data: getData(state)
})

const enhance = compose(
  withRouter,
  connect(mapStateToProps)
)

export default enhance(FooterContainer)
