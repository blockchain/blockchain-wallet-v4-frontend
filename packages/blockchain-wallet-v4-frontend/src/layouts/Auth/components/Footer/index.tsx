import React from 'react'
import { useSelector } from 'react-redux'
import { formValueSelector } from 'redux-form'
import styled from 'styled-components'

import { LOGIN_FORM } from 'data/auth/model'
import { getMagicLinkData, getProduct } from 'data/auth/selectors'
import { LoginSteps, PlatformTypes, ProductAuthOptions } from 'data/types'
import { media } from 'services/styles'

import DropdownLanguage from './DropdownLanguage'
import Help from './Help'
import InstitutionalPortal from './InstitutionalPortal'
import Version from './Version'

const FooterContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  width: 100%;
  margin: 6rem 0 2rem;

  ${media.mobile`
    flex-direction: column;
    margin-top: 0.5rem;
  `}
`

const FooterInner = styled.div`
  padding: 2rem 2rem 0;

  > * {
    margin: 0 0.5rem;
  }
`

const Footer = ({ path }) => {
  const authProduct = useSelector(getProduct)
  const step = useSelector((state) => formValueSelector(LOGIN_FORM)(state, 'step')) as LoginSteps
  const platform = useSelector(getMagicLinkData)?.platform_type

  const isMobile = platform !== PlatformTypes.IOS && platform !== PlatformTypes.ANDROID
  const showInstitutionalPortal =
    authProduct === ProductAuthOptions.EXCHANGE &&
    step !== LoginSteps.INSTITUTIONAL_PORTAL &&
    !isMobile &&
    path !== '/login'

  return (
    <FooterContainer>
      {showInstitutionalPortal && (
        <FooterInner>
          <InstitutionalPortal />
        </FooterInner>
      )}
      <FooterInner>
        <DropdownLanguage color='grey400' size='16px' />
        <Version />
        <Help />
      </FooterInner>
    </FooterContainer>
  )
}

export default Footer
