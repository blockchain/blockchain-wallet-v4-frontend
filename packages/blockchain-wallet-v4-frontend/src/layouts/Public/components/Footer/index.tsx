import React from 'react'
import styled from 'styled-components'

import { LoginSteps, ProductAuthOptions } from 'data/types'

import DropdownLanguage from './DropdownLanguage'
import Help from './Help'
import InstitutionalPortal from './InstitutionalPortal'
import Version from './Version'

const FooterInner = styled.div`
  padding: 2rem 2rem 0;
`

const Footer = (props) => (
  <>
    {props.authProduct === ProductAuthOptions.EXCHANGE &&
      props.formValues?.step !== LoginSteps.INSTITUTIONAL_PORTAL && (
        <FooterInner>
          <InstitutionalPortal />
        </FooterInner>
      )}
    <FooterInner>
      <DropdownLanguage color='grey400' size='16px' />
      <Version />
      <Help {...props} />
    </FooterInner>
  </>
)

export default Footer
