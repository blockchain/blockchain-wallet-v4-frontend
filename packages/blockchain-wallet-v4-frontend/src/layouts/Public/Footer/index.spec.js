import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Footer from './index'

jest.mock('components/DropdownLanguage', () => ({ DropdownLanguage: '' }))
jest.mock('blockchain-info-components', () => ({}))

describe('Public Footer Component', () => {
  window.APP_VERSION = 'v4.2.0'
  it('should match snapshot', () => {
    const component = shallow(<Footer />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
