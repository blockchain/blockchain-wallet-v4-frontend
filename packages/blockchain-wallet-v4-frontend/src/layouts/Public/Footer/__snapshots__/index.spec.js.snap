// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Public Footer Component should match snapshot 1`] = `
<Footer__Wrapper>
  <Footer__LinkContainer>
    <Component
      height="40px"
      name="blue-logo"
      width="40px"
    />
    <Component
      color="white"
      href="https://blockchain.com/explorer"
      size="12px"
      uppercase={true}
      weight={300}
    >
      <FormattedMessage
        defaultMessage="Data"
        id="layouts.public.footer.explorer"
        values={Object {}}
      />
    </Component>
    <Component
      color="white"
      href="https://blockchain.com/about"
      size="12px"
      uppercase={true}
      weight={300}
    >
      <FormattedMessage
        defaultMessage="About"
        id="layouts.public.footer.about"
        values={Object {}}
      />
    </Component>
    <Component
      color="white"
      href="https://blog.blockchain.com"
      size="12px"
      uppercase={true}
      weight={300}
    >
      <FormattedMessage
        defaultMessage="Blog"
        id="layouts.public.footer.blog"
        values={Object {}}
      />
    </Component>
    <Component
      color="white"
      href="https://support.blockchain.com"
      size="12px"
      uppercase={true}
      weight={300}
    >
      <FormattedMessage
        defaultMessage="Support"
        id="layouts.public.footer.support"
        values={Object {}}
      />
    </Component>
  </Footer__LinkContainer>
  <Footer__RightContainer>
    <[object Object]
      color="white"
    />
    <Component
      color="white"
      href="https://github.com/blockchain/blockchain-wallet-v4-frontend/releases"
      size="12px"
      target="_blank"
      weight={200}
    >
      Version 
      v4.2.0
    </Component>
  </Footer__RightContainer>
</Footer__Wrapper>
`;
